### ISSUE TITLE
Change default branch to `develop`

### ISSUE THREAD

**Issue by adamlm** [usdot-fhwa-stol/cooperative-perception-core#3]
### Types of Issue

- [ ] Anomaly report (something appears to not work correctly)
- [ ] Enhancement request (describe the enhancement being requested)
- [x] Other (please ensure the description clarifies why the issue doesn’t fall into either of the above categories)

### Descriptive summary

Type of issue: repository administration

The default branch for this repository is `main`, but it should be `develop` (which is also what other CARMA projects use). This is preventing issues from automatically closing when their associated PRs get merged. For example, issue #1 has a notice saying,

> This issue will close once commit `53c97fd` is merged into the `main` branch.

The issue should have closed when the commit was merged into `develop`.

For some context, here is a quote from the relevant [GitHub Docs](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#about-linked-issues-and-pull-requests)

> When you merge a linked pull request into the default branch of a repository, its linked issue is automatically closed.

### Carma version where this issue was discovered

N/A

### Expected behavior
N/A

### Actual behavior

N/A

### Steps to reproduce the actual behavior

N/A

### Related work

N/A
